### PR TITLE
#88: add delete protections

### PR DESCRIPTION
**Pull request by reece**
*Issue #88*

*Description of changes:*

This PR provides two protections against inadvertent deletions:

1. `--delete` is now required to delete users and groups. (A single flag controls both behaviors.)
2. Groups with a name beginning "AWS" will never be deleted because they likely correspond to Control Tower administrative groups.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by nandubatchu** [comment]
Hey @reece - I was not aware of this and accidentally ended up deleting all my control tower groups - is there any way to recover them?

**Comment by reece** [comment]
I was able to recreate the control tower groups by renrolling each account. Good luck. My heart dropped when I discovered that the control tower groups had been deleted.

File: internal/sync.go [hunk 1]
```diff
@@ -449,17 +453,28 @@ func (s *syncGSuite) SyncGroupsUsers(query string) error {
 
 		log := log.WithFields(log.Fields{"group": awsGroup.DisplayName})
 
+		// In mid-2022, AWS started using the prefix "AWS" for administrative
+		// purposes. Without this, ssosync deletes these administrative groups.
+		if awsGroup.DisplayName[:3] == "AWS" {
+			log.Warn("Refusing to delete")
```
Comment (sonrai-doyle) [inline internal/sync.go:459]:
```suggestion
			log.Warn("Refusing to delete group with the prefix 'AWS'")
```

**Comment by thapakazi** [comment]
> I was able to recreate the control tower groups by renrolling each account. Good luck. My heart dropped when I discovered that the control tower groups had been deleted, but it all worked out okay in the end.

even if those are delete, why would one need sso for control tower ?

**Comment by reece** [comment]
To use Google as the identity provider for all AWS access.

**Comment by reece** [comment]
Closing due to lack of response from code maintainers.

**Pull request closed without merging**